pkg/controllers/scheduling/schedule.go

```diff
@@ -253,19 +269,51 @@ func getWeights(defaultWeight map[string]int32, placement *clusterapiv1alpha1.Pl
 	}
 }
 
-func mergeWeights(defaultWeight map[string]int32, customizedWeight []clusterapiv1alpha1.PrioritizerConfig) map[string]int32 {
-	weights := make(map[string]int32)
+func mergeWeights(defaultWeight map[clusterapiv1alpha1.ScoreCoordinate]int32, customizedWeight []clusterapiv1alpha1.PrioritizerConfig) map[clusterapiv1alpha1.ScoreCoordinate]int32 {
+	weights := make(map[clusterapiv1alpha1.ScoreCoordinate]int32)
 	// copy the default weight
-	for k, v := range defaultWeight {
-		weights[k] = v
+	for sc, w := range defaultWeight {
+		weights[sc] = w
 	}
-	// override the default weight
+
 	for _, c := range customizedWeight {
-		weights[c.Name] = c.Weight
+		if c.ScoreCoordinate != nil {
+			weights[*c.ScoreCoordinate] = c.Weight
+		} else {
+			// override default weight
+			sc := clusterapiv1alpha1.ScoreCoordinate{
+				Type:    clusterapiv1alpha1.ScoreCoordinateTypeBuiltIn,
+				BuiltIn: c.Name,
+			}
+			weights[sc] = c.Weight
+		}
 	}
 	return weights
 }
 
+// Generate prioritizers for the placement.
+func getPrioritizers(weights map[clusterapiv1alpha1.ScoreCoordinate]int32, handle plugins.Handle) map[clusterapiv1alpha1.ScoreCoordinate]plugins.Prioritizer {
+	result := make(map[clusterapiv1alpha1.ScoreCoordinate]plugins.Prioritizer)
+	for k, v := range weights {
+		if v == 0 {
+			continue
+		}
+		if k.Type == clusterapiv1alpha1.ScoreCoordinateTypeBuiltIn {
+			switch {
+			case k.BuiltIn == PrioritizerBalance:
+				result[k] = balance.New(handle)
+			case k.BuiltIn == PrioritizerSteady:
+				result[k] = steady.New(handle)
+			case strings.HasPrefix(k.BuiltIn, PrioritizerResourcePrefix):
```

Comment:
the previous code only allowed two types.  I would avoid expanding the list in this PR

Reply:
yes, replace this line with PrioritizerResourceAllocatableCPU and PrioritizerResourceAllocatableMemory.